File: 1-25/07.py
#!/usr/bin/python
import math
def prime_check(x):
    if x == 2: return True
    if x % 2 == 0: return False
    return not any(x % i == 0 for i in range(2, int(math.sqrt(x))+2))
count = 0 ; x = 2 ; plist = []
while count <= 10001:
    if prime_check(x):
        plist.append(x)
        count += 1
    x += 1
print(plist[10000])

File: Perl/07.pl
#!/usr/bin/perl
sub prime
{
    for (my $i = 3; $i < sqrt($_[0]) + 1; $i++) {
        if ($_[0] % $i == 0) { return 0; }
    }
    return 1;
}
my $pcount = 1, $n = 3;
while ($pcount < 10001) {
    if (prime($n)) { $pcount++; }
    $n += 2;
}
print $n - 2;
